### PR TITLE
documentation: add experimental features flag guide to README

### PR DESCRIPTION
**Pull request by superlopuh**
Adds missing documentation to #3450

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ optimization capabilities.
 - [xDSL Developer Setup](#xdsl-developer-setup)
   - [Developer Installation](#developer-installation)
   - [Testing](#testing)
-  - [Formatting](#formatting)
+  - [Formatting](#formatting-and-typechecking)
```
Comment (alexarice) [inline README.md:34]:
```suggestion
  - [Formatting and Typechecking](#formatting-and-typechecking)
```

**Review by alexarice — APPROVED** [review]

**Comment by alexarice** [comment]
What does the warning look like? never seen this markdown feature before

**Comment by superlopuh** [comment]
It's a GH markdown feature, here's the new README from this PR: https://github.com/xdslproject/xdsl/blob/2b8fe6a1e5780333d2f8364e9178df0a8aad93e3/README.md

And here's the doc: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

File: README.md [hunk 2]
```diff
@@ -135,6 +135,20 @@ Furthermore, all python code must run through [pyright](https://github.com/micro
 without errors. Pyright can be run on all staged files through the
 makefile using `make pyright`.
 
+> [!IMPORTANT]
+>
+> ## Experimental Pyright Features
```
Comment (alexarice) [inline README.md:140]: This looks like it's being promoted to a top-level heading in the important block

File: README.md [hunk 3]
```diff
@@ -135,6 +135,20 @@ Furthermore, all python code must run through [pyright](https://github.com/micro
...
+> [!IMPORTANT]
+>
+> #### Experimental Pyright Features
+>
+> xDSL currently relies on an experimental feature of Pyright called TypeForm
+> TypeForm is [in discussion](https://discuss.python.org/t/pep-747-typeexpr-type-hint-for-a-type-expression/55984) and will likely land in some future version of Python.
+>
+> For xDSL to type check correctly using Pyright, please add this to your `pyproject.toml`:
+>
+> ``` text
```
Comment (alexarice) [inline README.md:147]:
```suggestion
> ```yaml
```
or at least delete the space?

Comment (alexarice) [reply README.md]: Not sure if you wanted the real-time review

Comment (alexarice) [reply README.md]: My bad this should have been toml not yaml

File: README.md [hunk 4]
```diff
@@ -135,6 +135,20 @@ Furthermore, all python code must run through [pyright](https://github.com/micro
...
+> [!IMPORTANT]
+>
+> #### Experimental Pyright Features
+>
+> xDSL currently relies on an experimental feature of Pyright called TypeForm
+> TypeForm is [in discussion](https://discuss.python.org/t/pep-747-typeexpr-type-hint-for-a-type-expression/55984) and will likely land in some future version of Python.
+>
+> For xDSL to type check correctly using Pyright, please add this to your `pyproject.toml`:
+>
+> ```yaml
```
Comment (alexarice) [inline README.md:147]:
```suggestion
> ```toml
```

**Review by compor — APPROVED** [review]

**Pull request merged**